versioning/lockfiles/build_order.rst

```diff
@@ -0,0 +1,211 @@
+    $ conan export app1 app1/0.1@user/testing
+    $ conan export app2 app2/0.1@user/testing
+
+Now we will create a lockfile that captures the dependency graph for ``app1/0.1@user/testing``.
+In the same way we created lockfiles for a local ``conanfile.py`` in a user folder, we can also
+create a lockfile for a recipe in the Conan cache, with the ``--reference`` argument:
+
+.. code-block:: bash
+
+    $ conan lock create --reference=app1/0.1@user/testing --lockfile-out=app1.lock
+
+The resulting *app1.lock* lockfile will not be able to completely lock the binaries because such
+binaries do not exist at all. This can be checked in the *app1.lock* file, the packages do not
+contain a package revision (``prev``) field at all:
+
+.. code-block:: text
+
+     {
+        ...
+        "4": {
+        "ref": "pkga/0.1@user/testing",
+        "options": "",
+        "package_id": "5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9",
+        "context": "host"
+        }
+        ...
+     }
+
+We can now compute the "build-order" of the dependency graph. The "build-order" lists
+in order all the packages that needs to be built from sources. As the lockfile contains
+the information of existing locked package binaries, the logic is the following:
+
+- If a package is fully locked (it contains a package revision field ``prev`` in the lockfile),
+  it cannot be built from sources, so it will never be returned
+- If a package is not fully locked (it does not contain a package revision ``prev`` in the lockfile),
+  the package will be returned. This situation happens both when the package binary doesn't exist yet,
+  or when the ``--build`` argument was used while creating the lockfile.
+
+.. code-block:: bash
+
+    $ conan lock build-order app1.lock --json=build_order.json
+
+The resulting *build_order.json* file is a list of lists, structured by levels of possible parallel builds:
+
+.. code-block:: text
+
+    [
+      # First level pkga
+      [["pkga/0.1@user/testing", "5ab8...1ac9", "host", "4"]],
+      # Second level pkgb and pkgc
+      [["pkgb/0.1@user/testing", "cfd1...ec23", "host", "3"],
+       ["pkgc/0.1@user/testing", "cfd1...ec23", "host", "5"]],
+      # Third level pkgd
+      [["pkgd/0.1@user/testing", "d075...5b9d", "host", "2"]],
+      # Fourth level pkgd
+      [["app1/0.1@user/testing", "3bf2...5188", "host", "1"]]
+    ]
+
+Every item in the outer list is a "level" in the graph, a set of packages that needs to be built, and
+are independent to every other package in the level, so they can be built in parallel. Levels in the
+build order must be respected, the second level cannot be started until the first level has finished
+and so on. In this example, once the build of ``pkga/0.1@user/testing`` finishes, as it is the only
+item in the first level, the second level can start, and it can build in parallel both ``pkgb/0.1@user/testing``
+and ``pkgc/0.1@user/testing``, because, as we can see in the graph, they are independent. It is necessary
```

Comment:
In parallel at the end and avoid repetition
```suggestion
item in the first level, the second level can start, and it can build both ``pkgb/0.1@user/testing`` and ``pkgc/0.1@user/testing`` in parallel. It is necessary
```